software/bedtools/slop/main.nf

```diff
@@ -0,0 +1,47 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+params.options = [:]
+def options    = initOptions(params.options)
+
+process BEDTOOLS_SLOP {
+    tag "$meta.id"
+    label 'process_medium'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), publish_id:meta.id) }
+
+    conda (params.enable_conda ? "bioconda::bedtools =2.29.2" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/bedtools:2.29.2--hc088bd4_0"
+    } else {
+    container "quay.io/biocontainers/bedtools:2.29.2--hc088bd4_0"
+    }
+
+
+    input:
+        tuple val(meta), path(beds), path (sizes)
+
+    output:
+        tuple val(meta), path("*.slop.bed"), emit: slopbed
+        path  "*.version.txt", emit: version
+
+    script:
+        def software = getSoftwareName(task.process)
+        def prefix   = options.suffix ? "${meta.id}${options.suffix}" : "${meta.id}"
+        def header = params.header ? "-header":''
+        def pct = params.pct ? "-pct":''
+
+        def symmetry = ''
+        if (meta.symmetry) {
+
+        """
+        slopBed -i $beds -g $sizes -b $params.b $header $pct $options.args> ${prefix}.slop.bed
```

Comment:
What is `params.b` here?

Reply:
It is to provide the integer value for -b ( Increase the BED/GFF/VCF entry by the same number base pairs in each direction) - would you want the slop process to require integer values for -l and -r? As of now, the process takes symmetry as a meta.id and then refers to either -l,-r or -b based on that. Do you have any recommendations for a more efficient way of going about this?

Reply:
👍🏽 Any parameter that isn't to do with an input/output, setting resources, or related to the sample should be provided via `options.args` which would be evaluated in the pipeline. This just means the module will be more usable by others too.

Reply:
Thank you for the clarification! I will make sure to fix that!